pkg/filesystem/image.go

```diff
@@ -136,6 +131,24 @@ func (fs *FileSystem) generateThumbnail(ctx context.Context, file *model.File) e
 	src := ""
 	if conf.SystemConfig.Mode == "slave" || file.GetPolicy().Type == "local" {
 		src = file.SourceName
+	} else {
```

Comment:
This does not seems to be a good abstraction:

1. Generator specific logic should keeps in `thumb` package. This `image.go` should be generator-agnostic.

2. `src` parameter in thumb generator is originally for file path. I understand in ffmpeg it happens can also be file URL, but it will make the interface definition inconsistent.

Reply:
ok, I'll move it into `thumb` package, but this also means it needs sort of changing function sign too.